k8s/with-defaults.bzl

```diff
@@ -96,6 +100,7 @@ k8s_defaults = repository_rule(
         "context": attr.string(mandatory = False),
         "user": attr.string(mandatory = False),
         "namespace": attr.string(mandatory = False),
+        "kubectl_args": attr.string(mandatory = False),
```

Comment:
I do not think `attr.string` is the best type for this argument. how about `string_list`?

https://docs.bazel.build/versions/master/skylark/lib/attr.html#string_list

Reply:
Originally I thought this might do best in a hash/associative array (what do they call these in python?) but this is going to contain any number of flags which may or may not contain arguments. I.E.
```--chunk-size=100 --all-namespaces --show-kind=true```

In fact many of the flags themselves may or may not have arguments. all-namespaces can be passed as ```--all-namespaces=true``` or simply ```--all-namespaces```.

So this could go either way. Is there an attr.map or something?

That might give us even better control over the input in regards to escaping strings mentioned below. At some point the keys could be restricted to a list of the allowed arguments in kubectl (some automation to generate said list would obviously be required.)

Reply:
There is `attr.string_dict`, but my vote would still be for `string_list`.

My rule-of thumb:

* If the rule needs to do something clever with a particular flag, make it a regular attribute.
* If the flag is just passed to the underlying tool, try to do as little processing as possible and let the underlying tool validate it. `args` is pretty good for this.

Reply:
Looking through the code and reading a bit on unit testing in Python I don't see how to write unit tests for this. In no small part because almost none of the code I've added is in a standard python file but in .bzl files which I assume get transformed into actual python at some point? 

I see a couple of e2e-test.sh files, mostly under the examples directory. Are you thinking e2e?

My code is just a copy/paste and slight modification of existing code. If there were unit tests of the existing I would have done the same cribbing off the test. ^_^ Does anyone have examples I can look at?

If you're thinking the e2e stuff I've looked at those and I think I can muddle through copying one of the two hello examples.

Reply:
Thanks for taking the time to work on the tests. Unit testing skylark rules is not possible, so yes, we are thinking e2e tests. You should just be able to copy some of the examples (as you say) and add some kubectl_args and verify they behave as expected.